### PR TITLE
ceph-osd: add dmcrypt scenario

### PR DESCRIPTION
**Pull request by leseb**
add the ability to encrypt osd data store using dm-crypt

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

File: roles/ceph-osd/tasks/scenarios/dmcrypt.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# NOTE (leseb): the prepare process must be parallelized somehow...
+# if you have 64 disks with 4TB each, this will take a while
+# since Ansible will sequential process the loop
+
+# NOTE (alahouze): if the device is a partition, the parted command below has
+# failed, this is why we check if the device is a partition too.
+- name: automatic prepare osd disk(s) without partitions
+  command: ceph-disk prepare --dmcrypt --cluster "{{ cluster }}" "/dev/{{ item.key }}"
+  register: prepared_osds
```
Comment (dang) [inline roles/ceph-osd/tasks/scenarios/dmcrypt.yml]: Why register?  The output isn't being used anywhere?

Comment (leseb) [inline roles/ceph-osd/tasks/scenarios/dmcrypt.yml]: Good catch, I don't know :)

**Comment by leseb** [comment]
That's a tough one, I hope I'm not breaking anything

**Comment by leseb** [comment]
test this please

**Comment by leseb** [comment]
test this please

**Pull request merged**